src/libraries/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray_1.Builder.cs

```diff
@@ -345,6 +345,33 @@ public void AddRange(ImmutableArray<T> items, int length)
                 }
             }
 
+            /// <summary>
+            /// Adds the specified items to the end of the array.
+            /// </summary>
+            /// <param name="items">The items.</param>
+            public void AddRange(ReadOnlySpan<T> items)
+            {
+                int offset = this.Count;
+                this.Count += items.Length;
+
+                items.CopyTo(new Span<T>(_elements, offset, items.Length));
+            }
+
+            /// <summary>
+            /// Adds the specified items to the end of the array.
+            /// </summary>
+            /// <param name="items">The items.</param>
+            public void AddRange<TDerived>(ReadOnlySpan<TDerived> items) where TDerived : T
+            {
+                int offset = this.Count;
+                this.Count += items.Length;
+
+                for (int i = 0; i < items.Length; i++)
+                {
+                    _elements[offset + i] = items[i];
+                }
```

Comment:
```suggestion
                Span<T> elements = _elements.AsSpan(offset, items.Length);
                for (int i = 0; i < items.Length; i++)
                {
                    elements[i] = items[i];
                }
```

Reply:
Fixed. Just confirm: main purpose is to avoid arithmetic (offset + i) during each iteration, right ?